src/internet_identity/tests/integration/latest_delegation_origins.rs

```diff
@@ -0,0 +1,115 @@
+//! Tests for tracking the latest delegation origins.
+
+use canister_tests::api::internet_identity as api;
+use canister_tests::flows;
+use canister_tests::framework::{
+    env, install_ii_canister, install_ii_canister_with_arg, principal_1, upgrade_ii_canister,
+    II_WASM,
+};
+use ic_cdk::api::management_canister::main::CanisterId;
+use ic_test_state_machine_client::{CallError, StateMachine};
+use internet_identity_interface::internet_identity::types::{AnchorNumber, InternetIdentityInit};
+use serde_bytes::ByteBuf;
+use std::time::Duration;
+
+const DAY_SECONDS: u64 = 24 * 60 * 60;
+
+/// Verifies that origins are recorded.
+#[test]
+fn should_record_used_delegation_origin() -> Result<(), CallError> {
+    let env = env();
+    let canister_id = install_ii_canister(&env, II_WASM.clone());
+    let user_number = flows::register_anchor(&env, canister_id);
+
+    delegation_for_origin(&env, canister_id, user_number, "https://some-dapp.com")?;
+
+    let latest_origins = api::stats(&env, canister_id)?.latest_delegation_origins;
+    assert_eq!(latest_origins, vec!["https://some-dapp.com".to_string()]);
+    Ok(())
+}
+
+/// Verifies that the configured limit for the number of latest used origins is respected.
+#[test]
+fn should_record_max_delegation_origins() -> Result<(), CallError> {
+    let env = env();
+    let canister_id = install_ii_with_low_latest_origin_limit(&env);
```

Comment:
nit: maybe pass the limit as an argument, so that the code below (the for-loop, and the asserts) don't have to depend on the constant used in the helper, but rather one can use a constant local to the test?